### DISCUSSION TITLE
Identifier/authority language

### DISCUSSION THREAD

**Discussion by jtauber** [bible-technology/scripture-burrito#232] [Next]
This is a first attempt to get some language out about identifiers and authorities. I'll continue to refine this but wanted to put it out here for discussion. This is intended to be a conceptual overview, informed by use cases and informing the spec itself. Much of this language can make its way to the documentation.

> A Scripture Burrito is a representation of a resource or project at a particular point in time.
>
> The Scripture Burrito metadata provides a means to indicate a machine-readable identifier given to that resource or project by a particular authority.
> 
> In many cases this authority is a piece of software but it could be an individual or organization. "Authority" here is just the entity responsible for minting the identifier.
> 
> This authority need not be the creator of the burrito or its underlying resource or project. I need not even have had any involvement in the burrito's content. An authority giving an identifier to a resource or project represented by a burrito is not a claim of authorship or ownership. A resource or project may have any number of authorities giving it identifiers but each authority should (or must?) give a resource or project only one identifier.
> 
> The authority is responsible for ensuring it does not give the same identifier to two resources or projects it considers distinct (however, see separate discussion on revision management). In other words, identifiers need only be unique scoped within an authority. However, the combination of an authority + identifier must be globally unique. Authorities themselves must be identified by URI.
>
> There is no assumption that the structure of an identifier is meaningful to anyone but the authority minting it. Other than the authority + identifier combination being a globally unique identifier, there is no assumption it is useful for any purpose other than identification. In particular, it is not assumed that an arbitrary system could use the identifier to resolve the origin of a burrito, retrieve other versions, etc. It is possible that a system does have knowledge of a particular authority (especially if it is that authority) in order to perform such actions but that is outside the scope of the identifer system.

Also see #233 on versions.

**Comment by joelthe1** [comment]
Just a note to add verbiage around who and when gets to change what in the identifiers.

**Reply by jtauber** [reply]
added this line:

> An authority may not alter the identification information provided by another authority other than to remove it entirely if the content of the burrito is changed.

Let me know if that suffices.

**Reply by jonathanrobie** [reply]
> The Scripture Burrito metadata provides a means to indicate a machine-readable identifier given to the burrito by a particular authority.

This doesn't seem right to me.  The identifier is not the identifier of the Burrito, it is the identifier of the resource or project. A Burrito is just an exchange format.

When I read this section, I think it tells me that DBL is an authority that knows what the identifier '2880c78491b2f8ce' corresponds to, DBL has created a revision 91, and whoever created this Burrito thinks they are giving me content that corresponds to this thing DBL knows about.   If I give someone over at DBL a call and ask them to send the same content to someone else,  they might create a new SB, but it will still correspond to this resource.

```
  "idAuthorities": {
    "dbl": {
      "id": "https://thedigitalbiblelibrary.org",
      "name": {
        "en": "The Digital Bible Library"
      }
    }
  },
  "identification": {
    "systemId": {
      "dbl": {
        "id": "2880c78491b2f8ce",
        "revision": "91"
      }
    },
```

**Reply by jtauber** [reply]
> The identifier is not the identifier of the Burrito, it is the identifier of the resource or project.

I'll re-work things to make this clearer. I'm glad this discussion is drawing out some very important concepts to clarify in the text.

**Comment by jtauber** [comment]
I have updated the text at the top to refer to the identifier as identifying the resource or project and the burrito being a representation of a particular resource or project.
